### DISCUSSION TITLE
How do I avoid nodes with the same name but different weights or labels

### DISCUSSION THREAD

**Discussion by hrshy0629** [networkx/networkx#6189] [Q&A] [answered]
I have some questions.
First, when I process the people's social network graph. I found a lot of the same names. However, the source code of add_node will update different people but not create a new node for different people. How can I avoid the problem?
Second, it consumes a lot of time and memory when I look for partial paths in the directed acyclic graph. However, the source code adopts the DFS algorithm, which takes much time in a big graph. What I want to know is, is there a better way to do it for directed acyclic graphs?

**Comment by boothby** [comment, marked as answer]
The label of a node must be a unique identifier.  Without knowing your dataset, it's hard to make a recommendation, but often social media sites have a username different from the display name, or even an ID number.  Without a unique identifier, there would be no way to know which person a particular edge belongs to.

As for you path finding problem, please share the code that you're using.
